modules/policy_document/main.tf

```
@@ -1,17 +1,19 @@
 locals {
   empty = data.aws_iam_policy_document.empty.json
 
+  templates = flatten([var.template != null ? [var.template] : [], var.templates])
   # first check the template paths for the template. if found, apply the template
   # vars. if the template does not exist at that path, use an empty policy document
   # instead.
   policy_documents = [
-    for path in var.template_paths : fileexists("${path}/${var.template}") ? (
-      templatefile(
-        "${path}/${var.template}",
-        var.template_vars
-      )
-    ) : local.empty
-  ]
+    for template in local.templates : [
+      for path in var.template_paths : fileexists("${path}/${local.templates}") ? (
+        templatefile(
+          "${path}/${var.template}",
+          var.template_vars
+        )
+      ) : local.empty
+  ]]
```

wrap this in a `flatten()`. because there is a nested looped list in a list, the result is `[ [ { objects... } ] ]`, when what we want is `[ { objects... } ]`.

```suggestion
    for template in local.templates : [
      for path in var.template_paths : fileexists("${path}/${local.templates}") ? (
        templatefile(
          "${path}/${var.template}",
          var.template_vars
        )
      ) : local.empty
  ]]
```